### PR TITLE
Change use localized field label checkbox to button

### LINKED ISSUE #4341: 'Use localized field labels' checkbox in App Resource should be a button
From @maxpatiiuk comment: 

> Given that the "Use localized field labels" checkbox is reloading the page, it would be more semantically correct to have that as a button.
> I know that it's reused with the Form Meta menu, so we could convert it to a button there too.
> If it's a button, the label would have to be different depending on on/off state

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4341

### CODE REVIEW

File: specifyweb/frontend/js_src/lib/localization/forms.ts [hunk 1]
```diff
@@ -986,6 +986,12 @@ export const formsText = createDictionary({
     'uk-ua': 'Використовуйте локалізовані назви полів',
     'de-ch': 'Verwenden Sie lokalisierte Feldbezeichnungen',
   },
+  localizedLabels: {
+    'en-us': 'Localized Labels',
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/localization/forms.ts:990]: since this is used as a button label, can you reword it to be something like `Display localized field labels` and `Display database field names` 

(could change `Display` to `Use` if you want)

Comment (maxpatiiuk) [reply specifyweb/frontend/js_src/lib/localization/forms.ts:990]: also, if `useFieldLabels` is no longer used, then remove that string

File: specifyweb/frontend/js_src/lib/components/FormEditor/Editor.tsx [hunk 2]
```diff
@@ -197,10 +196,11 @@ function UseLabelsSchema(): JSX.Element {
   };
 
   return (
-    <Label.Inline>
-      <Input.Checkbox checked={useFieldLabels} onValueChange={update} />
-      {formsText.useFieldLabels()}
-    </Label.Inline>
+    <Button.Secondary onClick={update}>
+      {useFieldLabels
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/components/FormEditor/Editor.tsx:200]: Shouldn't these labels be swapped?

i.e if we currently use localized field labels, then the button should say `Use Database Field Names` so that the user knows what action to expect when they click the button

**Review by maxpatiiuk — CHANGES_REQUESTED** [review]

File: specifyweb/frontend/js_src/lib/components/FormEditor/Editor.tsx (hunk 2 above)
Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/FormEditor/Editor.tsx:200]: yes indeed!

**Comment by CarolineDenis** [comment]
I'm not convinced of the button in the meta menu dialog as the reload only happens once the user closes the dialog.

**Review by emenslin — APPROVED** [review]
Looks good, I think a button here makes a lot more sense. 
Screenshot 2024-01-04 102436

**Review by pashiav — APPROVED** [review]
Looks good!

https://github.com/specify/specify7/assets/118556884/f5c9f42a-9e22-49bc-8575-189bd9c5080b

File: specifyweb/frontend/js_src/lib/localization/forms.ts [hunk 3]
```diff
@@ -986,6 +986,12 @@ export const formsText = createDictionary({
     'uk-ua': 'Використовуйте локалізовані назви полів',
     'de-ch': 'Verwenden Sie lokalisierte Feldbezeichnungen',
   },
+  showLocalizedLabels: {
```
Comment (maxpatiiuk) [inline specifyweb/frontend/js_src/lib/localization/forms.ts:989]: (optional) to better match the naming of the `useFieldLabels` option, could rename this to `showFieldLabels` and `showDataModelLabels`

**Review by maxpatiiuk — APPROVED** [review]
